pyTigerGraph/pyTigerGraphBase.py

```diff
@@ -188,6 +183,23 @@ def __init__(self, host: str = "http://127.0.0.1", graphname: str = "MyGraph",
             self.gsUrl = self.host + ":" + self.gsPort
         self.url = ""
 
+        if self.username.startswith("arn:aws:iam::"):
+            import boto3
+            from botocore.awsrequest import AWSRequest
+            from botocore.auth import SigV4Auth
+            # Prepare a GetCallerIdentity request.
+            request = AWSRequest(
+                method="POST",
+                url="https://sts.amazonaws.com/?Action=GetCallerIdentity&Version=2011-06-15",
```

Comment:
Is this url going to be true for all customers or just one? The Identity version of 2011 is pretty old...

Reply:
Yeah, it is aws' api version https://docs.aws.amazon.com/STS/latest/APIReference/API_GetCallerIdentity.html